Mac M1 chntpw not building (Openssl 1.0 requirement not satisfied)

Hello.

I have a mac M1 mba and I am trying to install sidneys/homebrew/chntpw

i have tried via rosetta and also via regular arm homebrew. unfortunately there is an issue building openssl 1.0 for this cpu,

there seems to be some instructions to build it manually on openssl github repo, any chance this can be incorporated into your script?

I have same problem.

```bash

x86_64cpuid.s:273:7: error: invalid token in expression
 cmpq $0,%rax
      ^
x86_64cpuid.s:273:7: error: invalid operand
 cmpq $0,%rax
      ^
x86_64cpuid.s:274:9: error: unknown token in expression
 cmoveq %rcx,%rax
        ^
x86_64cpuid.s:274:9: error: invalid operand
 cmoveq %rcx,%rax
        ^
make[1]: *** [x86_64cpuid.o] Error 1
make: *** [build_crypto] Error 1

Do not report this issue to Homebrew/brew or Homebrew/core!

> > I have the same issue. Anyone got it fixed?
> 
> To get it fixed, use @itsmattkc's PR, then download chntpw from sidney's cask, here: https://github.com/sidneys/chntpw/archive/0.99.6.tar.gz
> 
> Untar the downloaded file, edit the Makefile replace the line that contains `OSSLPATH` by this one:
> 
> ```
> OSSLPATH=/opt/homebrew/Cellar/openssl@1.0/1.0.2u
> ```
> 
> Save the Makefile, go to the terminal, navigate to the folder where chntpw is located, execute `make` and, finally, `cp chntpw /opt/hombrew/bin` and you're good to go

but osslpath is '/opt/homebrew/Cellar/openssl@1.1/1.1.1k` to me,i have get a issue `suhualin@MacBook chntpw-0.99.6 % make
gcc -c -DUSEOPENSSL -g -I. -I/opt/homebrew/Cellar/openssl@1.1/1.1.1k/include -Wl chntpw.c
warning: unknown warning option '-Wl' [-Wunknown-warning-option]
chntpw.c:141:34: error: expected expression
        DES_set_odd_parity((des_cblock *)key);
                                        ^
chntpw.c:141:22: error: use of undeclared identifier 'des_cblock'
        DES_set_odd_parity((des_cblock *)key);
                            ^
chntpw.c:186:3: error: unknown type name 'des_key_schedule'; did you mean 'DES_key_schedule'?
  des_key_schedule ks;
  ^~~~~~~~~~~~~~~~
  DES_key_schedule
/opt/homebrew/Cellar/openssl@1.1/1.1.1k/include/openssl/des.h:43:3: note: 'DES_key_schedule' declared here
} DES_key_schedule;
  ^
chntpw.c:187:3: error: unknown type name 'des_cblock'; did you mean 'DES_cblock'?
  des_cblock deskey;
  ^~~~~~~~~~
  DES_cblock
`

> curl -LO https://gist.github.com/minacle/e9dedb8c17025a23a453f8f30eced3da/raw/908b944b3fe2e9f348fbe8b8800daebd87b5966c/openssl@1.0.rb curl -LO https://gist.github.com/minacle/e9dedb8c17025a23a453f8f30eced3da/raw/908b944b3fe2e9f348fbe8b8800daebd87b5966c/chntpw.rb brew install --formula --build-from-source ./openssl@1.0.rb brew install --formula --build-from-source ./chntpw.rb rm ./openssl@1.0.rb ./chntpw.rb
> 
> >

=> Downloading https://github.com/sidneys/chntpw/archive/0.99.6.tar.gz
Already downloaded: /Users/kuldeepaher/Library/Caches/Homebrew/downloads/dee692592b6fc0d6db43012210cf70e9ef7a6018e20818b5f15187cbd3a06e39--chntpw-0.99.6.tar.gz
==> Patching
==> Applying chntpw-0.99.6-opthomebrew.patch
patching file Makefile
==> make
Last 15 lines from /Users/kuldeepaher/Library/Logs/Homebrew/chntpw/01.make:
2021-12-21 07:18:33 +0000

make

gcc -c -DUSEOPENSSL -g -I. -I/opt/homebrew/Cellar/openssl@1.0/1.0.2u/include -Wl chntpw.c
gcc -c -DUSEOPENSSL -g -I. -I/opt/homebrew/Cellar/openssl@1.0/1.0.2u/include -Wl ntreg.c
gcc -c -DUSEOPENSSL -g -I. -I/opt/homebrew/Cellar/openssl@1.0/1.0.2u/include -Wl edlib.c
gcc -c -DUSEOPENSSL -g -I. -I/opt/homebrew/Cellar/openssl@1.0/1.0.2u/include -Wl cpnt.c
gcc -c -DUSEOPENSSL -g -I. -I/opt/homebrew/Cellar/openssl@1.0/1.0.2u/include -Wl reged.c
gcc -DUSEOPENSSL -g -I. -I/opt/homebrew/Cellar/openssl@1.0/1.0.2u/include -Wl -o cpnt cpnt.o -L/opt/homebrew/Cellar/openssl@1.0/1.0.2u/lib /opt/homebrew/Cellar/openssl@1.0/1.0.2u/lib/libcrypto.a
clang: error: no such file or directory: '/opt/homebrew/Cellar/openssl@1.0/1.0.2u/lib/libcrypto.a'
make: *** [cpnt] Error 1
make: *** Waiting for unfinished jobs....

Do not report this issue to Homebrew/brew or Homebrew/core!


I'm getting this issue... after running the easy installation sxript

--- a/Formula/openssl@1.0.rb
+++ b/Formula/openssl@1.0.rb
@@ -6,16 +6,16 @@ class OpensslAT10 < Formula
   sha256 "ecd0c6ffb493dd06707d38b14bb4d8c2288bb7033735606569d8f90f89669d16"
   version_scheme 1
 
-  patch do
-    url "https://gist.githubusercontent.com/felixbuenemann/5f4dcb30ebb3b86e1302e2ec305bac89/raw/b339a33ff072c9747df21e2558c36634dd62c195/openssl-1.0.2u-darwin-arm64.patch"
-    sha256 "4ad22bcfc85171a25f035b6fc47c7140752b9ed7467bb56081c76a0a3ebf1b9f"
-  end
-
   bottle do
   end
 
   keg_only :provided_by_macos,
     "openssl/libressl is provided by macOS so don't link an incompatible version"
+
+  patch do
+    url "https://gist.githubusercontent.com/felixbuenemann/5f4dcb30ebb3b86e1302e2ec305bac89/raw/b339a33ff072c9747df21e2558c36634dd62c195/openssl-1.0.2u-darwin-arm64.patch"
+    sha256 "4ad22bcfc85171a25f035b6fc47c7140752b9ed7467bb56081c76a0a3ebf1b9f"
+  end
 
   # SSLv2 died with 1.1.0, so no-ssl2 no longer required.
   # SSLv3 & zlib are off by default with 1.1.0 but this may not

--- a/Formula/openssl@1.0.rb
+++ b/Formula/openssl@1.0.rb
@@ -6,6 +6,11 @@ class OpensslAT10 < Formula
   sha256 "ecd0c6ffb493dd06707d38b14bb4d8c2288bb7033735606569d8f90f89669d16"
   version_scheme 1
 
+  patch do
+    url "https://gist.githubusercontent.com/felixbuenemann/5f4dcb30ebb3b86e1302e2ec305bac89/raw/b339a33ff072c9747df21e2558c36634dd62c195/openssl-1.0.2u-darwin-arm64.patch"
+    sha256 "4ad22bcfc85171a25f035b6fc47c7140752b9ed7467bb56081c76a0a3ebf1b9f"
+  end
+
   bottle do
   end
 
@@ -37,7 +42,7 @@ class OpensslAT10 < Formula
     # Whilst our env points to opt_bin, by default OpenSSL resolves the symlink.
     ENV["PERL"] = Formula["perl"].opt_bin/"perl" if which("perl") == Formula["perl"].opt_bin/"perl"
 
-    arch_args = %w[darwin64-x86_64-cc enable-ec_nistp_64_gcc_128]
+    arch_args = %W[darwin64-#{Hardware::CPU.arch}-cc enable-ec_nistp_64_gcc_128]
 
     ENV.deparallelize
     system "perl", "./Configure", *(configure_args + arch_args)

